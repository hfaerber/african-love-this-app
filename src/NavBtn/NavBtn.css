.button-nav {
  width: 18%;
  background-color: #ef9e69;
  font-size: 2em;
  margin: 2% 1%;
  border: 3px ridge #f28242;
  border-radius: 10px;
  cursor: pointer;
}
