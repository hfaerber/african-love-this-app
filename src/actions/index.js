export const updateUser = user => ({
  type: 'UPDATE_USER',
  user
});

export const updateCountries = countries => ({
  type: 'UPDATE_COUNTRIES',
  countries
});

export const toggleStudyStatus = name => ({
  type: 'TOGGLE_STUDY_STATUS',
  name
});

export const updateSelectedFilter = filter => ({
  type: 'UPDATE_SELECTED_FILTER',
  filter
});

export const updateSearchQuery = query => ({
  type: 'UPDATE_SEARCH_QUERY',
  query
});

export const updateError = errorMessage => ({
  type: 'UPDATE_ERROR',
  errorMessage
});

export const updateLoadingStatus = status => ({
  type: 'UPDATE_LOADING_STATUS',
  status
});
