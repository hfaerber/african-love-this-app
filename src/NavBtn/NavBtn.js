import React from 'react';
import { Link } from 'react-router-dom';
import './NavBtn.css';
import { updateUser } from '../actions';
import { connect } from 'react-redux';

export const NavBtn = ({ text, link, user, updateUser }) => {
  return(
    <Link to={link}>
      <button type='button' className='button-nav' id={text} 
        onClick={() => updateUser(user)}>
        {text}
      </button>
    </Link>
  )
}

export const mapDispatchToProps = dispatch => ({
  updateUser: user => dispatch(updateUser(user))
})

export default connect(null, mapDispatchToProps)(NavBtn);
