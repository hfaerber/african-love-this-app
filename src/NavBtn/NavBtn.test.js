import React from 'react';
import { NavBtn, mapDispatchToProps } from './NavBtn';
import { updateUser } from '../actions';
import { shallow } from 'enzyme';

describe('NavBtn Container', () => {
  describe('NavBar component', () => {
    let wrapper, mockUser, mockText, mockUpdateUser;

    beforeEach(() => {
      mockUser = 'Ben';
      mockText = 'Study';
      mockUpdateUser = jest.fn();
      wrapper = shallow(<NavBtn
          text={mockText}
          user={mockUser}
          updateUser={mockUpdateUser}
        />);
    })

// add snapshot tests

    // it('should invoke updateUser on click', () => {
    //
    // })
  })

  describe('mapDispatchToProps', () => {
    it('should call dispatch with a user action when updateUser is called',
    () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = updateUser('Ben');
      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.updateUser('Ben');
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
})
