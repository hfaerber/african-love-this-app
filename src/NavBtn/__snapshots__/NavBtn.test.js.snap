// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBtn Container should match snapshot 1`] = `
<NavLink
  className="button-nav"
  to="/study"
>
  Study
</NavLink>
`;
